csharp/src/Ice/RequestFailedException.cs

```diff
@@ -10,58 +10,54 @@ namespace Ice
 
     public partial class RequestFailedException
     {
-        protected virtual string DefaultIceMessage
+        protected virtual string DefaultMessage
             => Facet.Length == 0 ? $"request for operation `{Operation}' on Ice object `{Id}' failed"
                 : $"request for operation `{Operation}' on Ice object `{Id}' with facet `{Facet}' failed";
 
-        public override string Message => IceMessage.Length > 0 ? IceMessage : DefaultIceMessage;
+        public override string Message => HasCustomMessage ? base.Message : DefaultMessage;
 
-        public RequestFailedException(Identity id, string facet, string operation,
-                                      System.Exception innerException)
-            : base(innerException)
+        public RequestFailedException(string message, Identity id, string facet, string operation,
+                                      Exception innerException)
+            : base(message, innerException)
         {
             Id = id;
             Facet = facet;
             Operation = operation;
-            IceMessage = "";
         }
     }
```

Comment:
Why don't we generate this constructor?

Reply:
Because many (possibly most) exceptions are not intended/expected to have an inner exception, but yes maybe we should generate these ctors just in case.

Reply:
Fixed - added generated constructor with message and innerException parameters.